### PR TITLE
8 create boilerplate

### LINKED ISSUE #8: Create boilerplate
## Description
As an team we want something to build upon so we can provide value for our stakeholder by working in parallel.

## Acceptance criteria
- [ ] There should exist a common ground / base implementation of the infrastructure.
- [x] A complete folder structure, the most core folders should exist.
- [x] A team member should be able to add a new page by creating a new file and referencing it in a single place.
- [ ] There is a set up navigation structure including side panel for navigation

## Dependencies
None.

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
The foundation of the project, including the entire project structure and some examples of implementations.

image
image

More details about the architecture will be written in #11.

Branch coverage is >80% in most of the projects, but since there is not a lot of example code it is not possible to reach 80% yet. When we implement more UI then it will be possible to reach that high of a coverage.

## Definition of done
- [x] All acceptance criteria are completed.
- [ ] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.

Closes #8

### CODE REVIEW

File: src/Infrastructure/InjectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public static class InjectionExtensions
+{
+    /// <summary>
+    ///     Registers infrastructure services.
+    /// </summary>
+    /// <param name="services">The existing services.</param>
+    /// <returns>The same service container.</returns>
+    public static IServiceCollection AddInfrastructure(this IServiceCollection services)
+    {
+        return services.AddSingleton<IWeatherRepository, WeatherRepository>();
```
Comment (JacobBredin) [inline src/Infrastructure/InjectionExtensions.cs:21]: Scary, I need to look into dependency injection for sure

**Review by JacobBredin — APPROVED** [review]
LGTM! Just two questions

**Review by Gigset — APPROVED** [review]
WIth my limitied knowledge and some help, I managed to run this on my own pc. Therefore I approve this PR.

File: src/Infrastructure/Weather/WeatherRepository.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        return Enumerable.Range(1, 5).Select(index =>
+                             new WeatherForecast
+                             {
+                                 Date = DateTime.Now.AddDays(index),
+                                 TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
+                                 Summary = _summaries[RandomNumberGenerator.GetInt32(_summaries.Length)],
+                             })
+                         .ToArray();
+    }
+}
```
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Infrastructure/Weather/WeatherRepository.cs:35]: Yeah, I'm going to write about this in the README. It's entirely based off of [this architecture](https://github.com/jasontaylordev/CleanArchitecture#overview). Take a look at the image I included in the PR as well.

Briefly, Application contains all our business logic. Infrastructure contains all our interactions with the outside world (e.g. database/file system access). `Application/Weather` implements how we use the weather data, while `Infrastructure/Weather` implements how we store the weather data. 

The general concept is that `Application` defines interfaces for things it needs, and `Infrastructure` implements them to connect to the outside world.

File: src/Infrastructure/InjectionExtensions.cs (hunk 1 above)
Comment (martinjonsson01) [reply src/Infrastructure/InjectionExtensions.cs:21]: Basically what is happening here is that `Infrastructure` is saying: "if you're looking for an instance of `IWeatherRepository` then here you go, have `WeatherRepository`."

[Here's a page that describes it in detail.](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/dependency-injection?view=aspnetcore-6.0)

File: src/Client/Shared/NavMenu.razor [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <button title="Navigation menu" class="navbar-toggler" @onclick="ToggleNavMenu">
+            <span class="navbar-toggler-icon"></span>
+        </button>
+    </div>
+</div>
+
+<div class="@NavMenuCssClass" @onclick="ToggleNavMenu">
+    <nav class="flex-column">
+        <div class="nav-item px-3">
+            <NavLink class="nav-link" href="example">
```
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Client/Shared/NavMenu.razor:13]: Ah, right. That's fair. The `FetchData` page was originally linked to here, but I removed the redundant example pages. I'll add in an empty page as an example of how to define the route to it.

**Pull request merged**